### PR TITLE
Jenkins lineendings

### PR DESCRIPTION
**Pull request by AdamBrousseau**

### CODE REVIEW

File: buildenv/jenkins/lineEndingsCheck [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    returnStdout: true
+                ).trim()
+
+                switch (TYPE) {
+                    case ~/empty/:
+                        echo "Empty file: '${it}'"
+                        break
+                    case ~/.*text.*/:
+                        switch (it) {
+                            case ~/.*\.bat|.*\.cmd/:
```
Comment (keithc-ca) [inline buildenv/jenkins/lineEndingsCheck]: The bash script I offered did a case-insensitive match for `*.bat` and `*.cmd` files. Does this do the same?

File: buildenv/jenkins/lineEndingsCheck [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                                        echo "Good windows script: '${it}' type: '${TYPE}'"
+                                        break
+                                    default:
+                                        echo "ERROR - should have CRLF line terminators: '${it}' type: '${TYPE}'"
+                                        FAIL = true
+                                        BAD_FILES << new String("${it}")
+                                }
+                            default:
+                                switch (TYPE) {
+                                    case ~/.*CRLF line terminators.*/:
```
Comment (keithc-ca) [inline buildenv/jenkins/lineEndingsCheck]: The `LF` in the pattern should be `.*` to catch CR-only line-ends.

File: buildenv/jenkins/lineEndingsCheck [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                }
+                echo "${HASH}"
+                echo "${HASHES}"
+                sh 'exit 1'
+            } else {
+                echo "All modified files appear to have correct line endings"
+            }
+        }
+    }
+}
```
Comment (keithc-ca) [inline buildenv/jenkins/lineEndingsCheck:96]: I almost want to suggest that we should reject files without a final line-end (like this!).

File: buildenv/jenkins/lineEndingsCheck [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+String[] BAD_FILES
+String HASH = '#'
+String HASHES = '###################################'
+
+stage('Line Endings Check') {
+    node ('master') {
+        timestamps {
+            git url: SRC_REPO
+            sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+            sh "git checkout ${sha1}"
```
Comment (keithc-ca) [inline buildenv/jenkins/lineEndingsCheck]: The `--detach` option might suppress some of the useless noise including `You are in 'detached HEAD' state`.

File: buildenv/jenkins/lineEndingsCheck [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Boolean FAIL = false
+String SRC_REPO = 'https://github.com/AdamBrousseau/openj9.git'
+String[] BAD_FILES
+String HASH = '#'
+String HASHES = '###################################'
+
+stage('Line Endings Check') {
+    node ('master') {
+        timestamps {
```
Comment (keithc-ca) [inline buildenv/jenkins/lineEndingsCheck:31]: What is the meaning of `timestamps` here?

File: buildenv/jenkins/lineEndingsCheck (hunk 2 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/lineEndingsCheck]: so this?
`case ~/.*CR.* line terminators.*/:`

File: buildenv/jenkins/lineEndingsCheck (hunk 3 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/lineEndingsCheck:96]: Not sure what you mean here? Are you suggesting I add an empty line after L96?

File: buildenv/jenkins/lineEndingsCheck (hunk 4 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/lineEndingsCheck]: I can add `--detach`

File: buildenv/jenkins/lineEndingsCheck (hunk 5 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/lineEndingsCheck:31]: Jenkins pipeline syntax. Everything inside the timestamps block will output to Jenkins console with timestamps at the start of each line.
More info https://wiki.jenkins.io/display/JENKINS/Timestamper

File: buildenv/jenkins/lineEndingsCheck (hunk 1 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/lineEndingsCheck]: Added .toLowerCase() on the switch of the filename

File: buildenv/jenkins/lineEndingsCheck (hunk 5 above)
Comment (keithc-ca) [reply buildenv/jenkins/lineEndingsCheck:31]: Thanks.

File: buildenv/jenkins/lineEndingsCheck (hunk 3 above)
Comment (keithc-ca) [reply buildenv/jenkins/lineEndingsCheck:96]: I meant you should finish line 96 (by adding a newline).

File: buildenv/jenkins/lineEndingsCheck (hunk 2 above)
Comment (keithc-ca) [reply buildenv/jenkins/lineEndingsCheck]: Yes, that's what I meant.

**Pull request closed without merging**